test/extended/util/framework.go

```diff
@@ -219,7 +221,20 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface,
 		defer w.Stop()
 
 		for {
-			val, ok := <-w.ResultChan()
+			timeout := make(chan bool, 1)
+			go func() {
+				time.Sleep(10 * time.Second)
```

Comment:
@gabemontero could you please clarify why we're waiting for at most 10s on reading from the watcher?

Shouldn't we just have a timeout on the whole operation (the 15m so far)?